graphql/issuedNumber/resolver.ts

```diff
@@ -68,22 +69,41 @@ const IssuedNumberResolver = {
       return appointments[0];
     },
     cancelTurn: async (parent, args: MutationCancelTurnArgs, ctx: Context) => {
-      let appointments = await getPendingTurns(args.clientId, args.shopId, ctx);
+      if (!ctx.tokenInfo?.isValid) {
+        return new ApolloError('Invalid token', 'INVALID_TOKEN');
+      }
 
-      if (!appointments.length) {
-        return new ApolloError(
-          'There is no pending appointment.',
-          'APPOINtMENT_NOT_EXISTS'
-        );
+      if (!ctx.tokenInfo?.clientId) {
+        return new ApolloError('No client id', 'NO_CLIENT_ID');
+      }
+
+      const id = decodeId(args.turnId);
+
+      if (!id) {
+        return new ApolloError('Invalid id', 'INVALID_TURN_ID');
+      }
+
+      const issuedNumber = await ctx.prisma.issuedNumber.findOne({
```

Comment:
instead of finding the turn and then update it, perhaps you can just call update searching by turnId, clientId and shopId